server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processor/AstProcessor.java

```diff
@@ -39,6 +39,7 @@ public class AstProcessor {
    */
   public List<SyntaxError> processSyntaxTree(ProcessingContext ctx, Node rootNode) {
     for (ProcessingPhase phase : ProcessingPhase.values()) {
+      ThreadInterruptionUtil.checkThreadInterrupted();
```

Comment:
Added more interrupt checks , so that analysis could close computation quicker based on the current thread interruption check

Reply:
Does it make a difference, given it is called anytime the process function is entered?